Unify doc mark-ups

muzimuzhi opened the pull request:
Mark-up all commands (`gh help issue`), flags (`--json`), environment variables (`GH_PATH`), and paths to files found in doc strings with backticks consistently. Currently various delimiters are used: `"..."`, `'...'`, `` `...` ``, and no delimiters.

- All cases found in raw strings in `` long: heredoc.Doc(`...`) ``, `` Long: heredoc.Doc(`...`) ``, and `` Long: heredoc.Docf(`...`, "`") `` are adapted.
- Some types of special words are changed to be surrounded by double-quotes: scopes (`"repo"`), special values ("POST"), patterns ("OWNER/REPO").
- Line wraps are _not_ adjusted, partially because I find the max character number per line is inconsistent.

muzimuzhi committed:
Wrap cli flags in backticks uniformly

muzimuzhi committed:
Wrap cli commands in backticks uniformly

muzimuzhi committed:
Misc docs improvements

muzimuzhi committed:
Wrap env variables in backticks uniformly

muzimuzhi committed:
Wrap path to file in backticks

muzimuzhi committed:
Add leftovers

muzimuzhi committed:
Proof-reading

cliAutomation commented:
Hi! Thanks for the pull request. Please ensure that this change is linked to an issue by mentioning an issue number in the description of the pull request. If this pull request would close the issue, please put the word 'Fixes' before the issue number somewhere in the pull request body. If this is a tiny change like fixing a typo, feel free to ignore this message.

andyfeller commented:
@muzimuzhi : Thanks for your efforts to improve the CLI!  I believe I understand the changes you have made and see what impact this would have regarding the formatting of help usage within the CLI and on the website.

As per our [contributing guidelines](https://github.com/cli/cli/blob/ba60f89f424967605f327348ff436f68d24d775d/.github/CONTRIBUTING.md) we ask that PRs are created only for pre-approved work. Often there is more than one outcome for a solution and it helps us a lot to understand what our users are trying to do without muddying PRs with loosely related discussions.

**Could I trouble you with creating this issue while we review these changes for any concerns or additional ideas please?** 🙇

andyfeller commented on pkg/cmd/alias/set/set.go:
```diff
@@ -35,21 +35,21 @@ func NewCmdSet(f *cmdutil.Factory, runF func(*SetOptions) error) *cobra.Command
 
-			If the expansion starts with "!" or if "--shell" was given, the expansion is a shell
-			expression that will be evaluated through the "sh" interpreter when the alias is
+			If the expansion starts with "!" or if %[1]s--shell%[1]s was given, the expansion is a shell
```
Might as well include `!` too:

```suggestion
			If the expansion starts with %[1]s!%[1]s or if %[1]s--shell%[1]s was given, the expansion is a shell
```

andyfeller commented on pkg/cmd/alias/imports/import.go:
```diff
@@ -49,10 +49,10 @@ func NewCmdImport(f *cmdutil.Factory, runF func(*ImportOptions) error) *cobra.Co
 
 			Use "-" to read aliases (in YAML format) from standard input.
```
```suggestion
			Use %[1]s-%[1]s to read aliases (in YAML format) from standard input.
```

andyfeller commented on pkg/cmd/alias/set/set.go:
```diff
@@ -35,21 +35,21 @@ func NewCmdSet(f *cmdutil.Factory, runF func(*SetOptions) error) *cobra.Command
 			Define a word that will expand to a full gh command when invoked.
 
 			The expansion may specify additional arguments and flags. If the expansion includes
 			positional placeholders such as "$1", extra arguments that follow the alias will be
```
```suggestion
			positional placeholders such as %[1]s$1%[1]s, extra arguments that follow the alias will be
```

andyfeller commented on pkg/cmd/api/api.go:
```diff
@@ -75,7 +75,7 @@ func NewCmdApi(f *cmdutil.Factory, runF func(*ApiOptions) error) *cobra.Command
 
 			Placeholder values "{owner}", "{repo}", and "{branch}" in the endpoint
```
```suggestion
			Placeholder values %[1]s{owner}%[1]s, %[1]s{repo}%[1]s, and %[1]s{branch}%[1]s in the endpoint
```

andyfeller replied:
One small point I can't leave a comment/suggestion on being outside of change boundaries might be switching `"graphql"` too as it is command invoked here.

andyfeller commented on pkg/cmd/extension/command.go:
```diff
@@ -300,7 +300,7 @@ func NewCmdExtension(f *cmdutil.Factory) *cobra.Command {
+					The repository argument can be specified in "OWNER/REPO" format as well as a full URL.
 					The URL format is useful when the repository is not hosted on github.com.
 
 					To install an extension in development from the current directory, use "." as the
```
We should probably update this, too

```suggestion
					To install an extension in development from the current directory, use %[1]s.%[1]s as the
```

andyfeller commented on pkg/cmd/extension/command.go:
```diff
@@ -441,16 +441,16 @@ func NewCmdExtension(f *cmdutil.Factory) *cobra.Command {
 					Press q to quit.
 
-					Running this command with --single-column should make this command
+					Running this command with %[1]s--single-column%[1]s should make this command
```
I can't leave a comment due to change boundaries, but we should probably wrap `?` and `q`

andyfeller commented on pkg/cmd/release/download/download.go:
```diff
@@ -46,13 +46,13 @@ func NewCmdDownload(f *cmdutil.Factory, runF func(*DownloadOptions) error) *cobr
-                        is required.
-		`),
+			latest release in the project. In this case, %[1]s--pattern%[1]s or %[1]s--archive%[1]s
+			is required.
```
Nice catch!

muzimuzhi replied:
Just to be sure, did you mean the space-to-tab restoration?

andyfeller commented on pkg/cmd/root/help_topic.go:
```diff
@@ -31,75 +31,75 @@ var HelpTopics = []helpTopic{
+			%[1]sNO_COLOR%[1]s: set to any value to avoid printing ANSI escape sequences for color output.
 
-			CLICOLOR: set to "0" to disable printing ANSI colors in output.
+			%[1]sCLICOLOR%[1]s: set to "0" to disable printing ANSI colors in output.
```
```suggestion
			%[1]sCLICOLOR%[1]s: set to %[1]s0%[1]s to disable printing ANSI colors in output.
```

muzimuzhi commented:
@andyfeller https://github.com/cli/cli/issues/8348 is created

andyfeller commented on pkg/cmd/root/help_topic.go:
```diff
@@ -31,75 +31,75 @@ var HelpTopics = []helpTopic{
+			%[1]sCLICOLOR%[1]s: set to "0" to disable printing ANSI colors in output.
 
-			CLICOLOR_FORCE: set to a value other than "0" to keep ANSI colors in output
+			%[1]sCLICOLOR_FORCE%[1]s: set to a value other than "0" to keep ANSI colors in output
```
```suggestion
			%[1]sCLICOLOR_FORCE%[1]s: set to a value other than %[1]s0%[1]s to keep ANSI colors in output
```

andyfeller reviewed (changes requested):
Firstly, I appreciate the level of effort to review all of these commands and align formatting, spacing issues, and typos/wordage! 🙇 

I've left a few suggestions for additional places that might/should be quoted to follow this style.  Let me know what you think and whether these suggestions are consistent/inconsistent with what you found while doing this work. 👍

muzimuzhi commented:
@andyfeller Thanks for your careful review! I was aware of (all but `graphql`) cases you pointed out (and struggled with them for quite some time). The line I drawn when opening this PR was

> * Some types of special words are changed to be surrounded by double-quotes: scopes (`"repo"`), special values ("POST"), patterns ("OWNER/REPO").

(I classified value `"0"`, argument(?) `"-"` and keys pressed like `"q"` all to type "special values".)

I'm open to changing all such cases, and I'm also willing to wait to see what feedbacks #8348 will receive.

andyfeller commented:
> @andyfeller Thanks for your careful review! I was aware of (all but `graphql`) cases you pointed out (and struggled with them for quite some time). The line I drawn when opening this PR was
> 
> > * Some types of special words are changed to be surrounded by double-quotes: scopes (`"repo"`), special values ("POST"), patterns ("OWNER/REPO").
> 
> (I classified value `"0"`, argument(?) `"-"` and keys pressed like `"q"` all to type "special values".)
> 
> I'm open to changing all such cases, and I'm also willing to wait to see what feedbacks #8348 will receive.

@muzimuzhi : I can understand the struggle!!!

In an effort to be as objective as possible, how about we take a look at [GitHub Docs](https://docs.github.com/en) for similar examples to help guide how to handle:

1. Scopes
2. HTTP methods
3. Patterns
4. Key shortcuts

If the GitHub Docs highlights them with code blocks / fencing, then we should feel confident of doing the same.  As such, let me see about finding examples and will update the issue with links for consideration.

muzimuzhi commented:
GitHub Docs (https://docs.github.com/en / https://github.com/github/docs) itself is not 100% consistent on formatting special phrases.

> 1. Scopes

Inconsistent
- ``The "admin:enterprise" scope``
  https://github.com/github/docs/blob/main/content/admin/identity-and-access-management/provisioning-user-accounts-for-enterprise-managed-users/configuring-scim-provisioning-for-enterprise-managed-users.md
- ``with the **admin:enterprise** scope``
  https://github.com/github/docs/blob/main/content/admin/identity-and-access-management/provisioning-user-accounts-for-enterprise-managed-users/configuring-scim-provisioning-with-okta.md
- ``with the `admin:enterprise` scope``
  https://github.com/github/docs/blob/main/content/admin/identity-and-access-management/using-saml-for-enterprise-iam/configuring-authentication-and-provisioning-for-your-enterprise-using-okta.md

> 2. HTTP methods

``send a `POST` request to``
https://docs.github.com/en/actions/deployment/protecting-deployments/creating-custom-deployment-protection-rules#approving-or-rejecting-deployments

> 3. Patterns

``If the `OWNER/` portion of the `OWNER/REPO` repository``
https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository?tool=cli#cloning-a-repository

``In this example, replace `owner/repo` with the repository identifier.``
https://docs.github.com/en/codespaces/developing-in-a-codespace/creating-a-codespace-for-a-repository#creating-a-codespace-for-a-repository

> 4. Key shortcuts

A special tag `<kbd>` like `press <kbd>Shift</kbd>` which rendered as framed inline code.
https://docs.github.com/en/issues/planning-and-tracking-with-projects/managing-items-in-your-project/editing-items-in-your-project

For remaining types other than scopes, I think the suggestion from GitHub Docs is: use code blocks.

muzimuzhi commented:
> > 4\. Key shortcuts
> 
> A special tag `<kbd>` like `press <kbd>Shift</kbd>` which rendered as framed inline code. https://docs.github.com/en/issues/planning-and-tracking-with-projects/managing-items-in-your-project/editing-items-in-your-project

Also in italic `Press _Shift_`
https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/managing-contribution-settings-on-your-profile/viewing-contributions-on-your-profile#viewing-contributions-from-specific-times

muzimuzhi committed:
Merge branch 'trunk' into docs/unify-markup

muzimuzhi committed:
Wrap special values and placeholders like `-` and `$1`

muzimuzhi committed:
Wrap HTTP methods

muzimuzhi committed:
Leftovers

muzimuzhi committed:
Small docs tweaks

muzimuzhi committed:
Use "GitHub Actions" as a whole

muzimuzhi committed:
Update link to jq docs

muzimuzhi committed:
s/default branch/main branch/g

muzimuzhi committed:
Wrap scopes

muzimuzhi committed:
Add missing minimum required scope `gist`

muzimuzhi committed:
fixup! Use "GitHub Actions" as a whole

muzimuzhi commented:
I wrapped all cases, including scopes in backticks.

Changes to scopes are committed at last, so it's easy to drop them if the final decision is to not format them like inline code.

Some small doc fixing/improvements are also applied in separate commits. Feel free to ask me to propose them in separate PR(s) if that's the right way.

andyfeller commented:
> I wrapped all cases, including scopes in backticks.
> 
> Changes to scopes are committed at last, so it's easy to drop them if the final decision is to not format them like inline code.
> 
> Some small doc fixing/improvements are also applied in separate commits. Feel free to ask me to propose them in separate PR(s) if that's the right way.

@muzimuzhi : Excellent!  My apologies for the delayed follow up on my part with the holiday last week.  Let's get this re-reviewed and merged in, thank you so much for being a great contributor, Yukai! 🙇

andyfeller commented on pkg/cmd/actions/actions.go:
```diff
@@ -52,8 +52,8 @@ func actionsExplainer(cs *iostreams.ColorScheme) string {
 			To see more help, run 'gh help workflow <subcommand>'
```
In reviewing changes, I realized these commands should be quoted to.  I'm working on local changes to push to the branch as a way to address manual reviewed changes 👍

muzimuzhi replied:
Do they appear on https://cli.github.com? I didn't find them in web pages, but only in stdout of executing `gh actions`.

muzimuzhi replied:
Yes, currently `gh actions` is marked hidden from showing up in the list of available commands.

So this change works for future and it throws up a new question: although backtick marks are preferred on web, are they preferred in CLI stdout as well?

https://github.com/cli/cli/blob/b215d03e7b5990ea35bc0870a55d7b84f5d8c987/pkg/cmd/actions/actions.go#L13-L18

and from [doc for `cobra.Command`](https://pkg.go.dev/github.com/spf13/cobra@v1.6.1#Command)
```go
type Command struct {
	// ...
	// Hidden defines, if this command is hidden and should NOT show up in the list of available commands.
	Hidden bool
}
```

andyfeller commented on pkg/cmd/root/help_topic.go:
```diff
@@ -61,23 +61,23 @@ var HelpTopics = []helpTopic{
+			to, e.g. %[1]sless%[1]s.
 
 			%[1]sGLAMOUR_STYLE%[1]s: the style to use for rendering Markdown. See
 			<https://github.com/charmbracelet/glamour#styles>
```
Something I realized over the course of this PR is that URLs formatted with `<` and `>` are the only variants that get turned into links on the CLI manual website.

Rather than extending this PR further, that might be a great follow up to review that behavior and align it.

muzimuzhi replied:
issue tracking this: https://github.com/cli/cli/issues/8404

andyfeller commented on pkg/cmd/variable/delete/delete.go:
```diff
@@ -126,7 +126,7 @@ func removeRun(opts *DeleteOptions) error {
 			fmt.Fprintf(opts.IO.Out, "%s Deleted variable %s from %s environment on %s\n", cs.SuccessIconWithColor(cs.Red), opts.VariableName, envName, target)
 		} else {
-			fmt.Fprintf(opts.IO.Out, "%s Deleted Actions variable %s from %s\n", cs.SuccessIconWithColor(cs.Red), opts.VariableName, target)
+			fmt.Fprintf(opts.IO.Out, "%s Deleted GitHub Actions variable %s from %s\n", cs.SuccessIconWithColor(cs.Red), opts.VariableName, target)
```
Given line 127 doesn't call out the product name, I'm thinking we keep these more consistent.

```
			fmt.Fprintf(opts.IO.Out, "%s Deleted variable %s from %s\n", cs.SuccessIconWithColor(cs.Red), opts.VariableName, target)
```

andyfeller commented on pkg/cmd/variable/set/set.go:
```diff
@@ -219,7 +219,7 @@ func setRun(opts *SetOptions) error {
 			target += " environment " + envName
 		}
-		fmt.Fprintf(opts.IO.Out, "%s %s Actions variable %s for %s\n", cs.SuccessIcon(), result.Operation, result.Key, target)
+		fmt.Fprintf(opts.IO.Out, "%s %s GitHub Actions variable %s for %s\n", cs.SuccessIcon(), result.Operation, result.Key, target)
```
By removing the product name from the command above, I'm erring on the side of consistency here

```
		fmt.Fprintf(opts.IO.Out, "%s %s variable %s for %s\n", cs.SuccessIcon(), result.Operation, result.Key, target)
```

andyfeller committed:
Additional formatting changes

- gh help actions
  - highlight 3 commands to run for more information
- gh extension
  - wrapping url for online linking
  - fix install subcommand missing backtick formatting
- gh release create
  - fixing mixed indenting causing usage misalignment
- root help
  - backtick wrap help command for more info for consistency
- gh help environment
  - backtick wrap missed OWNER/REPO
- gh variable
  - brought consistency to output when deleting or setting variable

andyfeller reviewed (approved):
@muzimuzhi : I think this is ready to go!

I've manually built this branch and iterated over every command changed, catching a few things that remained including the footer added for help at the end of every command.

Once again, tremendous effort which I look forward to seeing on the CLI manual once shipped! 🎉

andyfeller commented on pkg/cmd/release/create/create.go:
```diff
@@ -94,11 +94,11 @@ func NewCmdCreate(f *cmdutil.Factory, runF func(*CreateOptions) error) *cobra.Co
 			git, push the tag to GitHub, then run this command.
 			Use %[1]s--notes-from-tag%[1]s to automatically generate the release notes
- 			from the annotated git tag.
+			from the annotated git tag.
```
Mixed spaces and tabs

andyfeller committed:
Fix lint warning

andyfeller reviewed (approved):
All manual testing looks good, will confirm website changes once release 👍

andyfeller committed:
For lint sake

YT026 reviewed (commented):
🧙🏻‍♂️

andyfeller commented:
@muzimuzhi : looking at https://cli.github.com/manual/gh_api versus https://web.archive.org/web/20231123042701/https://cli.github.com/manual/gh_api, sooooo much better!  thank you again for your diligence adding the polish our docs needed ❤️ 🙇

muzimuzhi commented:
@andyfeller Thank you too. It's only possible and made even better with your careful review.

pasamdk reviewed (commented):
S

pasamdk reviewed (commented):
H
